core/src/main/java/greencity/controller/UserController.java

```diff
@@ -410,10 +411,15 @@ public ResponseEntity<UserProfileStatisticsDto> getUserProfileStatistics(
         @ApiResponse(responseCode = "200", description = HttpStatuses.OK),
         @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST),
         @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(responseCode = "404", description = HttpStatuses.NOT_FOUND)
     })
     @GetMapping("/findByEmail")
     public ResponseEntity<UserVO> findByEmail(@RequestParam String email) {
-        return ResponseEntity.status(HttpStatus.OK).body(userService.findByEmail(email));
+        UserVO userVO = userService.findByEmail(email);
+        if (userVO == null) {
+            throw new NotFoundException(NotFoundException.class.getName());
```

Comment:
the message in `new NotFoundException` should reflect what was not found, find a matching one in `ErrorMessages` and use it here